### DISCUSSION TITLE
Remove CP Nav Items only for certain roles

### DISCUSSION THREAD

**Discussion by ARCHES-Academy** [statamic/cms#4509] [Q&A] [answered]
Is there a way to remove items from the CP Nav only if a user is a certain role?

I've tried testing with an `if` statement, but this throws an error:

``` php
if (User::current()->hasRole('teacher')) {
    Nav::extend(function ($nav) {
        $nav->remove('Content');
    });
}
```

And I've also tried using the `can()` method on the command, but it instead added a new section called "CAN" with the name of the permission I was testing for:
``` php
Nav::extend(function ($nav) {
    $nav->remove('Content')->can('permission here');
});
```

Is there another way to do this, or is this simply impossible with the current version?

**Comment by jasonvarga** [comment, marked as answer]
Try putting the `hasRole` check inside the `extend`.

```php
Nav::extend(function ($nav) {
  if (User::current()->hasRole('teacher')) {
        $nav->remove('Content');
  }
});
```

**Reply by ARCHES-Academy** [reply]
Awesome, that worked great! Thanks so much!
